lua/telescope/config.lua

```diff
@@ -187,12 +187,32 @@ local telescope_defaults = {
     - "absolute"  display absolute paths
     - "shorten"   only display the first character of each directory in
                   the path
+    - "function"  use a custom function defined in path_display_custom
```

Comment:
Not sure about the naming here. Function? callback?

Reply:
My thought was not to introduce a new config field but make `path_display` either a table or a function. 
So in `utils.transform_path` right at the beginning, we can do 
```lua
if type(path_display) == "function" then
  return path_display(opts, path) 
end
```

@l-kershaw what do you think

Reply:
Interesting... from a "make code simple", I like it. Is it confusing for users though? that one field can be either a table or function?

Reply:
I agree with @Conni2461. I think it is better to allow `path_display` to be either a table or a function. This means that we keep all of the configuration for displaying paths in one place.

I think the main thing is to ensure that we document it clearly. I think the general layout of the docs for `resolver.resolve_width` would be a good starting point, and then we can adjust from there.

Reply:
Sounds good. I'll make the change and we can fine tune it. Thanks.